### ISSUE TITLE
[bug][clickhouse]: select count(*) from numbers_mt(1) ATTEMPT_TO_READ_AFTER_EOF

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#4395]
**Summary**

clickhouse client:
```
Connecting to localhost:9000 as user default.
Connected to databend server version 2021.5.0 revision 54405.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.

databend :) select count(*) from numbers_mt(1);

SELECT count(*)
FROM numbers_mt(1)

Query id: 98f46f36-db0b-4a78-a164-6dbc96c44a80

Connecting to localhost:9000 as user default.
Connected to databend server version 2021.5.0 revision 54405.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.

Exception on client:
Code: 32. DB::Exception: Attempt to read after eof: while receiving packet from localhost:9000. (ATTEMPT_TO_READ_AFTER_EOF)

Connecting to localhost:9000 as user default.
Connected to databend server version 2021.5.0 revision 54405.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.
```

`select version()`:
```
databend :) select version();

SELECT version()

Query id: 8893ee32-791e-4a13-be43-440fe7cf66ca

Exception on client:
Code: 32. DB::Exception: Attempt to read after eof: while receiving packet from localhost:9000. (ATTEMPT_TO_READ_AFTER_EOF)

Connecting to localhost:9000 as user default.
Connected to databend server version 2021.5.0 revision 54405.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.

```

Server log(there is no error):
```
./target/debug/databend-query 
2022-03-11T07:24:12.258099Z  INFO databend_query: Config { config_file: "", query: QueryConfig { tenant_id: "admin", cluster_id: "", num_cpus: 16, mysql_handler_host: "127.0.0.1", mysql_handler_port: 3307, max_active_sessions: 256, clickhouse_handler_host: "127.0.0.1", clickhouse_handler_port: 9000, http_handler_host: "127.0.0.1", http_handler_port: 8000, http_handler_result_timeout_millis: 10000, flight_api_address: "127.0.0.1:9090", http_api_address: "127.0.0.1:8080", metric_api_address: "127.0.0.1:7070", http_handler_tls_server_cert: "", http_handler_tls_server_key: "", http_handler_tls_server_root_ca_cert: "", api_tls_server_cert: "", api_tls_server_key: "", api_tls_server_root_ca_cert: "", rpc_tls_server_cert: "", rpc_tls_server_key: "", rpc_tls_query_server_root_ca_cert: "", rpc_tls_query_service_domain_name: "localhost", table_engine_csv_enabled: false, table_engine_parquet_enabled: false, table_engine_memory_enabled: true, database_engine_github_enabled: true, wait_timeout_mills: 5000, max_query_log_size: 10000, table_cache_enabled: false, table_cache_snapshot_count: 256, table_cache_segment_count: 10240, table_cache_block_meta_count: 102400, table_memory_cache_mb_size: 256, table_disk_cache_root: "_cache", table_disk_cache_mb_size: 1024, management_mode: false, jwt_key_file: "" }, log: LogConfig { log_level: "INFO", log_dir: "./_logs" }, meta: {meta_address: "", meta_user: "", meta_password: "******"}, storage: StorageConfig { storage_type: "disk", disk: DiskStorageConfig { data_path: "_data", temp_data_path: "" }, s3: {s3.storage.region: "", s3.storage.endpoint_url: "https://s3.amazonaws.com", s3.storage.bucket: "", }, azure_storage_blob: {Azure.storage.container: "", } } }
2022-03-11T07:24:12.258569Z  INFO databend_query: DatabendQuery v-0.1.0-a145007-simd(1.61.0-nightly-2022-03-07T08:51:58.894099397+00:00)
2022-03-11T07:24:12.259522Z  INFO databend_query::catalogs::impls::mutable_catalog: use embedded meta
2022-03-11T07:24:12.269650Z  INFO databend_query: MySQL handler listening on 127.0.0.1:3307, Usage: mysql -uroot -h127.0.0.1 -P3307
2022-03-11T07:24:12.270390Z  INFO databend_query: ClickHouse handler listening on 127.0.0.1:9000, Usage: clickhouse-client --host 127.0.0.1 --port 9000
2022-03-11T07:24:12.273673Z  INFO databend_query: Http handler listening on 127.0.0.1:8000  examples:
curl --request POST '127.0.0.1:8000/v1/statement/' --header 'Content-Type: text/plain' --data-raw 'SELECT avg(number) FROM numbers(100000000)'
curl --request POST '127.0.0.1:8000/v1/query/' --header 'Content-Type: application/json' --data-raw '{"sql": "SELECT avg(number) FROM numbers(100000000)"}'
2022-03-11T07:24:12.274114Z  INFO poem::server: listening addr=socket://127.0.0.1:8000
2022-03-11T07:24:12.274971Z  INFO databend_query: Metric API server listening on 127.0.0.1:7070/metrics
2022-03-11T07:24:12.275071Z  INFO poem::server: server started
2022-03-11T07:24:12.275353Z  WARN databend_query::api::http_service: Http API TLS not set
2022-03-11T07:24:12.275360Z  INFO poem::server: listening addr=socket://127.0.0.1:7070
2022-03-11T07:24:12.276379Z  INFO poem::server: server started
2022-03-11T07:24:12.277959Z  INFO databend_query: HTTP API server listening on 127.0.0.1:8080
2022-03-11T07:24:12.278192Z  INFO poem::server: listening addr=socket://127.0.0.1:8080
2022-03-11T07:24:12.278281Z  INFO databend_query: RPC API server listening on 127.0.0.1:9090
2022-03-11T07:24:12.278801Z  INFO poem::server: server started
2022-03-11T07:24:12.281359Z  INFO databend_query: Databend query has been registered:"" to metasrv:[""].
2022-03-11T07:24:12.281776Z  INFO databend_query: Ready for connections.
2022-03-11T07:24:20.295863Z  INFO databend_query::servers::clickhouse::clickhouse_handler: ClickHouse connection coming: Ok(127.0.0.1:40516)
2022-03-11T07:24:20.645954Z  INFO databend_query::servers::clickhouse::clickhouse_handler: ClickHouse connection coming: Ok(127.0.0.1:40518)
```

Explain:
```
databend :) explain select count(*) from numbers_mt(1);

EXPLAIN
SELECT count(*)
FROM numbers_mt(1)

Query id: 4d62b47e-6f88-4aa1-9335-1791a73705c8

┌─explain─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ Projection: count():UInt64                                                                                                              │
│   Projection: 1 as count():UInt64                                                                                                       │
│     Expression: 1:UInt64 (Exact Statistics)                                                                                             │
│       ReadDataSource: scan schema: [dummy:UInt8], statistics: [read_rows: 1, read_bytes: 1, partitions_scanned: 1, partitions_total: 1] │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

Other query is ok.

**Comment by youngsofun** [comment]
`select current_user();` and `select 1` fail too

**Comment by sundy-li** [comment]
It's fixed in #4514
